docs/rfcs/XeTileandXeGPUDialect.md

```diff
@@ -0,0 +1,401 @@
+  %ret_value = XeGPU.atomic_rmw “addf” %value, %scatter_Desc1, %mask {mode = vc}: 
+          vector<16xbf16>, tensor_desc <16xbf16, #scattered>, vector<16xi1> to vector<16xbf16>
+``` 
+XeGPU.atomic_rmw reuses the arith dialect attribute, ::mlir::arith::AtomicRMWKindAttr. 
+PVC doesn’t support atomic operation on BF16/FP16 add. The BF16/FP16 matrix needs to be converted to FP32 to perform the reduction. 
+alloc_nbarrier allocates a number of named barriers. Named barrier is workgroup level resource, shared by all subgroups. 
+```mlir  	
+  XeGPU.alloc_nbarrier %nbarrier_count: i8
+```
+create_nbarrier assigns a role for a specific named barrier to be producer and/or consumer. The returned nbarrier object holds a description to the specified barrier, which encodes all the barrier information.  It also binds the current thread with the named barrier by holding the returned nbarrier object.  Multiple threads may bind to a same nbarrier so that they can sync with each other. 
+```mlir  
+  %nbarrier  = XeGPU.create_nbarrier  %nbarrier_id, %nbarrier_role, {num_producers = 2, num_consumers = 2} : i8, i8, nbarrier_role into nbarrier
+```
+enum class nbarrier_role : uint8_t {producer_consumer = 0,  producer = 1, consumer = 2 };
+
+nbarrier_arrive notifies other threads sharing the same named barrier that it has arrived. 
+```mlir   
+  XeGPU.nbarrier_arrive %nbarrier  
+```
+nbarrier_wait waits until all other threads sharing the same named barrier has signaled the arrival. 
+```mlir  
+  XeGPU. nbarrier_wait %nbarrier  
+```
+
+Mfence 
+Mfence is inserted to synchronize between write and following access. 
+```mlir  
+  XeGPU.mfence { {memory_kind = "ugm" , fence_op = "none", fence_scope = "local"}
+```
+    Fence_op: {"none",    "evict", "invalidate", "discard", "clean", "flushl3"};
+    Fence_scope : {"group", "local",  "tile",  "gpu",  "gpus",  "system", "sysacq"};
+    Memory_kind: {"ugm", "ugml", "tgm", "slm"};
+
+compile_hint passes performance hint to lower-level compiler. The schedule_barrier hint prevents instructions from being reordered by a lower-level compiler. For example, a prefetch instruction is location-sensitive, but the lower-level compiler may schedule it to an undesired location.  
+```mlir  
+XeGPU.compile_hint {hint=schedule_barrier}
+```
+nbarrrier, mfence, and compile_hint works on both VC mode and SIMT mode, since they access uniform values. 
+
+
+## Alternative
+
+The alternative design of tile data type is to reuse the memref data type. The memref data type needs to be enhanced to allow attributes. So the XeTile's tile data type can be expressed with memref associated with Tile attributes. XeTile.wg_map and XeTile.sg_map are examples of these attributes.    
+
+## Questions
+
+Currently there is no NVVM counterpart. XeGPU dialect uses SPIRV Intel extension to access joint-matrix or SPRIV external function to access intel GPU VC intrinsics. This may change in the future, so we expect XeGPU lowering may change accordingly.  
```

Comment:
Currently there is no lower-level GPU IR like NVVM available for Intel GPU compiler toolchain.